File: 프로그래머스/1/12912. 두 정수 사이의 합/두 정수 사이의 합.java
class Solution {
    public long solution(int a, int b) {
        long answer = 0;
        
        long start = Math.min(a, b);
        long end = Math.max(a, b);
        answer = (end - start +1) *(start+end)/2;
        return answer;
    }
}

File: 프로그래머스/1/12912. 두 정수 사이의 합/두 정수 사이의 합.kt
class Solution1 {
    fun solution(a: Int, b: Int): Long {
        val start = kotlin.math.min(a, b).toLong()
        val end = kotlin.math.max(a, b).toLong()
        return (start..end).sum()
    }
}

class Solution {
    fun solution(a: Int, b: Int): Long {
        val min = minOf(a, b)  // a와 b 중 작은 값
        val max = maxOf(a, b)  // a와 b 중 큰 값

        // min부터 max까지의 합을 계산
        return (min..max).sumOf { it.toLong() }
    }
}
